app/models/search.rb

```diff
@@ -24,11 +24,50 @@ def results
     res = res.bot_author unless bot_author.nil?
     res = res.unlabelled unless unlabelled.nil?
     res = res.is_private(is_private) unless is_private.nil?
+    res = apply_sort(res)
     res
   end
 
   private
 
+  def apply_sort(scope)
+    case sort_by
+    when 'subject'
+      scope.reorder("upper(notifications.subject_title) #{sort_order}")
+    when 'updated'
+      scope.reorder(updated_at: sort_order)
+    when 'read'
+      scope.reorder(last_read_at: sort_order)
```

Comment:
@paulleader `last_read_at` is often nil, not sure if that impacts this or not?

Reply:
@andrew Hrrrm, good point, on Postgres it sorts nil to be max, so in descending order you get all the unread ones first.

Reply:
Yeah, mysql treats null as minimum value :/

Will have a think about that and put in something tomorrow evening to make them consistent.

What would make more sense? The most sensible order for last read is probably descending (so most recent first) with unread ones at the end I would think.

Or, we could simply exclude unread ones from that search.